packages/next-auth/src/adapters.ts

```diff
@@ -84,7 +84,7 @@ export interface DefaultAdapter {
   getUserByAccount: (
     providerAccountId: Pick<AdapterAccount, "provider" | "providerAccountId">
   ) => Awaitable<AdapterUser | null>
-  updateUser: (user: Partial<AdapterUser>) => Awaitable<AdapterUser>
+  updateUser: (user: Partial<Omit<AdapterUser, "id">> & Pick<AdapterUser, "id">) => Awaitable<AdapterUser>
```

Comment:
This doesn't make much sense to me, it's the same type, no? 🤔
`Omit<AdapterUser, "id">> & Pick<AdapterUser, "id">`

Reply:
This is `AdapterUser` with `id` mandatory and all other fields optional